packages/eslint-plugin/src/rules/no-identifier-enum-member.ts

```diff
@@ -0,0 +1,46 @@
+import { createRule } from '../util';
+import { AST_NODE_TYPES } from '@typescript-eslint/experimental-utils';
+import { TSEnumMember } from '../../../typescript-estree/dist/ts-estree/ts-estree';
+
+export enum MessageId {
+  NoVariable = 'noVariable',
+}
+
+type Options = [];
+
+export default createRule<Options, MessageId>({
+  name: 'no-identifier-enum-member',
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description: 'Disallow identifier (aka variable) enum members',
+      category: 'Best Practices',
+      recommended: false,
+      requiresTypeChecking: false,
+    },
+    messages: {
+      [MessageId.NoVariable]: `Enum member can not be an ${AST_NODE_TYPES.Identifier}.`,
+    },
+    schema: [],
+  },
+  defaultOptions: [],
+  create(context) {
+    return {
+      TSEnumMember(node: TSEnumMember): void {
+        // If there is no initializer, then this node is just the name of the member, so ignore.
+        if (
+          node.initializer != null &&
+          node.initializer.type === AST_NODE_TYPES.Identifier
+        ) {
+          context.report({
+            node: node.id,
+            messageId: MessageId.NoVariable,
```

Comment:
Please use strings for messageIds instead of an enum

```suggestion
export default createRule({
  name: 'no-identifier-enum-member',
  meta: {
    type: 'suggestion',
    docs: {
      description: 'Disallow identifier (aka variable) enum members',
      category: 'Best Practices',
      recommended: false,
      requiresTypeChecking: false,
    },
    messages: {
      noVariable: `Enum member can not be an ${AST_NODE_TYPES.Identifier}.`,
    },
    schema: [],
  },
  defaultOptions: [],
  create(context) {
    return {
      TSEnumMember(node: TSEnumMember): void {
        // If there is no initializer, then this node is just the name of the member, so ignore.
        if (
          node.initializer != null &&
          node.initializer.type === AST_NODE_TYPES.Identifier
        ) {
          context.report({
            node: node.id,
            messageId: 'noVariable',
```

Reply:
I can do this, but I'm curious about why an enum isn't desirable.  It makes the dev and testing story a bit easier, after all.

Reply:
Mainly for consistency across the codebase. Every other rule uses string unions for this, so don't want to deviate for one rule.